src/m365/spo/commands/file/file-sharinglink-add.spec.ts

```diff
@@ -0,0 +1,241 @@
+    "grantedToIdentities": [],
+    "link": {
+      "scope": "organization",
+      "type": "edit",
+      "webUrl": "https://contoso.sharepoint.com/:b:/s/pnpcoresdktestgroup/EY50lub3559MtRKfj2hrZqoBWnHOpGIcgi4gzw9XiWYJ-A",
+      "preventsDownload": false
+    }
+  };
+
+  const graphResponseThree = {
+    "id": "2a021f54-90a2-4016-b3b3-5f34d2e7d932",
+    "roles": [
+      "read"
+    ],
+    "hasPassword": false,
+    "grantedToIdentitiesV2": [],
+    "grantedToIdentities": [],
+    "link": {
+      "scope": "anonymous",
+      "type": "embed",
+      "webUrl": "https://contoso.sharepoint.com/:b:/s/pnpcoresdktestgroup/EY50lub3559MtRKfj2hrZqoBWnHOpGIcgi4gzw9XiWYJ-A",
+      "preventsDownload": false
+    }
+  };
+
+  const graphResponseText: any = {
+    "id": "2a021f54-90a2-4016-b3b3-5f34d2e7d932",
+    "roles": "read",
+    "link": "https://contoso.sharepoint.com/:b:/s/pnpcoresdktestgroup/EY50lub3559MtRKfj2hrZqoBWnHOpGIcgi4gzw9XiWYJ-A",
+    "scope": "anonymous"
+  };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.post,
+      odata.getAllItems
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.FILE_SHARINGLINK_ADD);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('creates a sharing link from a file specified by the id', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${webUrl}/_api/web/GetFileById('${fileId}')?$select=SiteId,VroomItemId,VroomDriveId`) {
+        return fileDetailsResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/${fileDetailsResponse.SiteId}/drives/${fileDetailsResponse.VroomDriveID}/items/${fileDetailsResponse.VroomItemID}/createLink`) {
+        return graphResponseOne;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { webUrl: webUrl, fileId: fileId, output: 'json', type: 'view', scope: 'anonymous', verbose: true } } as any);
```

Comment:
output: 'json' is too verbose and can be removed here. This applies also to the other tests.